### PR TITLE
Update tests to be able to run them outside of AppVeyor

### PR DESCRIPTION
**Pull request by andreasjordan**
I want to run all the tests on a local lab using "real world" sql server instances.

I started with Get-DbaDb*, but try to get all tests working. Current target: getting *-DbaDb* working.

I learned that some test only work if the instance is on localhost. I think this is not very realistic as we promote to run dbatools on a central admin server, but sometime we need to move files around. Will try to get an overwiew and document this in the tests.

Will update this PR in the next days.

### CODE REVIEW

**Comment by andreasjordan** [comment]
Describe : Copy-DbaDbAssembly Integration Tests
Context  : 
Name     : It Error occurred in Describe block
Result   : Failed
Message  : Exception calling "Query" with "1" argument(s): "Exception calling "ExecuteWithResults" with "1" argument(s): "Execute with results failed for Database 'master'. ""

**Comment by andreasjordan** [comment]
Describe : Copy-DbaDbAssembly Integration Tests
Context  : 
Name     : It Error occurred in Describe block
Result   : Failed
Message  : Exception calling "Query" with "1" argument(s): "An exception occurred while executing a Transact-SQL statement or batch.Incorrect syntax near ','.
           Must declare the scalar variable "@assemblyName".
           Must declare the scalar variable "@hash"."

File: tests/Get-DbaDbCompatibility.Tests.ps1 [hunk 1]
```diff
@@ -14,14 +14,18 @@ Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
...
+    }
     Context "Gets compatibility for multiple databases" {
         $results = Get-DbaDbCompatibility -SqlInstance $script:instance1
         It "Gets results" {
             $results | Should Not Be $null
         }
         Foreach ($row in $results) {
-            It "Should return Compatiblity level of Version100 for $($row.database)" {
-                $row.Compatibility | Should Be "Version100"
+            It "Should return correct compatiblity level for $($row.database)" {
```
Comment (niphlod) [inline tests/Get-DbaDbCompatibility.Tests.ps1:27]: compat*I*bility here. may sound nitpicky, but given we're reviewing, we may fix "formalities" as well.

File: tests/Get-DbaDbCompatibility.Tests.ps1 [hunk 2]
```diff
@@ -32,8 +36,8 @@ Describe "$commandname Integration Tests" -Tags "IntegrationTests" {
         It "Gets results" {
             $results | Should Not Be $null
         }
-        It "Should return Compatiblity level of Version100 for $($results.database)" {
-            $results.Compatibility | Should Be "Version100"
+        It "Should return correct compatiblity level for $($results.database)" {
```
Comment (niphlod) [inline tests/Get-DbaDbCompatibility.Tests.ps1:39]: "compatibility" here as well

File: tests/Get-DbaDbDetachedFileInfo.Tests.ps1 [hunk 3]
```diff
@@ -16,6 +16,7 @@ Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
 Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
     BeforeAll {
         $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $versionName = $server.GetSqlServerVersionName()
```
Comment (niphlod) [inline tests/Get-DbaDbDetachedFileInfo.Tests.ps1:19]: didn't even know this existed, cool.

File: tests/Get-DbaDbDetachedFileInfo.Tests.ps1 [hunk 4]
```diff
@@ -39,7 +40,7 @@ Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
             $results.name | Should Be $dbname
         }
         It "Should be 2016" {
```
Comment (niphlod) [inline tests/Get-DbaDbDetachedFileInfo.Tests.ps1:42]: if this doesn't base itself around having a fixed version and checking it, this "IT" needs to be adjusted as well

File: tests/Get-DbaDbSharePoint.Tests.ps1 [hunk 5]
```diff
@@ -15,29 +15,47 @@ Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
 
 Describe "$commandname Integration Tests" -Tag "IntegrationTests" {
     BeforeAll {
+        $skip = $false
```
Comment (niphlod) [inline tests/Get-DbaDbSharePoint.Tests.ps1:18]: I never *really* got to use this function and I fear it spun up by potatoqualitee's period on working tirelessly on sharepoint farms. That being said I'm seeing multiple reports about either SQLPS loading DLLs that interfere with us and also DacFx-related issues (many cycling around core-noncore dilemma). That being said I trust you on this change, maybe @potatoqualitee  can chime in.

File: tests/Get-DbaDbSpace.Tests.ps1 [hunk 6]
```diff
@@ -24,30 +24,30 @@ Describe "$commandname Integration Tests" -Tags "IntegrationTests" {
...
     Context "Gets DbSpace when using -Database" {
         $results = Get-DbaDbSpace -SqlInstance $script:instance2 -Database $dbname
-        It -skip "Gets results" {
+        It "Gets results" {
             $results | Should Not Be $null
         }
         Foreach ($row in $results) {
-            It -skip "Should retreive space for $dbname" {
+            It "Should retreive space for $dbname" {
                 $row.Database | Should Be $dbname
```
Comment (niphlod) [inline tests/Get-DbaDbSpace.Tests.ps1:48]: maybe we should check if "space" is retrieved as well ? not a fixed value, of course, just "not null"

File: tests/New-DbaDbEncryptionKey.Tests.ps1 [hunk 7]
```diff
@@ -95,10 +95,14 @@ Describe "$CommandName Integration Tests for Async" -Tags "IntegrationTests" {
         }
     }
 
+    # TODO: I think I need some background on this. Was the intention to create the key or not to creeate the key?
```
Comment (niphlod) [inline tests/New-DbaDbEncryptionKey.Tests.ps1:98]: I concur with the "I don't know" bit

**Review by niphlod — CHANGES_REQUESTED** [review]
added some comments here and there, in general, LGTM

**Comment by andreasjordan** [comment]
Thanks for the comments. Will update the PR tomorrow.

File: tests/Get-DbaDbSharePoint.Tests.ps1 (hunk 5 above)
Comment (potatoqualitee) [reply tests/Get-DbaDbSharePoint.Tests.ps1:18]: i dont remember anything and also trust the change 😅

**Comment by andreasjordan** [comment]
Ok, I will stop here and not add any more commits to get this merged first.

But I can roll back individual files if the changes need more discussion.

**Comment by potatoqualitee** [comment]
beautiful! thank you both 🙇🏼  i will work on that dacfx issue hopefully this week or next.

**Pull request merged**